app/src/main/java/com/example/palletify/data/GeneratorData.kt

```diff
@@ -215,6 +216,63 @@ fun generateAnalogicPalette(
     return result;
 }
 
+fun generateMonochromePalette(
+    seeds: MutableSet<Palette.Color>,
+    count: Int
+): MutableList<Palette.Color> {
+    val rgbColors: MutableList<Array<Int>> = mutableListOf();
+
+    // if there are multiple seeds, average their rgb values and use that as a single seed
+    // (can change this later, not sure if it looks amazing)
+    var rSum = 0
+    var gSum = 0
+    var bSum = 0
+
+    for (color in seeds) {
+        val rgb = color.rgb;
+        rSum += rgb.r
+        gSum += rgb.g
+        bSum += rgb.b
+        rgbColors.add(arrayOf(rgb.r, rgb.g, rgb.b))
+    }
+    // set seed to averaged rgb values from all seeds
+    val averageRgbSeed = arrayOf(rSum / seeds.count(), gSum / seeds.count(), bSum / seeds.count())
+    val hslSeed = rgbToHsl(averageRgbSeed);
+
+    val countAsDouble = count.toDouble()
+    var currentCount: Double = rgbColors.count().toDouble()
+    val randomOffset = Random.nextDouble(0.05, 0.15)
+    // can adjust this
+    val step = 1
+
+    while (currentCount < countAsDouble) {
+        // keep hue the same, modify saturation and lightness by even step size based on # of colours
+        // add random offset to generate new palettes off of the same seeds
+        // S and L are both decimals from 0 to 1 so use modulo to bound
+        val newSaturation = (hslSeed[1] + step * currentCount / countAsDouble + randomOffset) % 1
+        val newLightness = (hslSeed[2] + step * currentCount / countAsDouble + randomOffset) % 1
+
+        // create new rgb colour
+        val newRgbColour = hslToRgb(arrayOf(hslSeed[0], newSaturation, newLightness))
+
+        // add to array
+        rgbColors.add(newRgbColour)
+        currentCount++
+    }
```

Comment:
is there a reason why you use a while loop instead of a for loop since this is a known size ?

you could also randomize the step

Reply:
No real reason, I can change it. I think randomizing the step instead of adding a random offset makes sense 👍